pom.xml

```diff
@@ -124,4 +166,83 @@
     </plugins>
   </reporting>
 
+  <profiles>
+    <profile> 
+      <id>release</id>
+      <build>
+        <plugins>
+          <plugin>
+            <!-- Package the source jar. -->
+            <groupId>org.apache.maven.plugins</groupId>
+            <artifactId>maven-source-plugin</artifactId>
+            <version>2.2.1</version>
+            <executions>
+              <execution>
+                <id>attach-sources</id>
+                <goals>
+                  <goal>jar-no-fork</goal>
+                </goals>
+              </execution>
+            </executions>
+          </plugin>
+          <plugin>
+            <!-- Package the javadoc jar. -->
+            <groupId>org.apache.maven.plugins</groupId>
+            <artifactId>maven-javadoc-plugin</artifactId>
+            <version>2.9.1</version>
+            <configuration>
+              <excludePackageNames>software.amazon.ion.apps:software.amazon.ion.impl</excludePackageNames>
+              <sourceFileExcludes>
+                <exclude>**/*Private*</exclude>
+              </sourceFileExcludes>
+              <overview>${project.basedir}/src/software/amazon/ion/overview.html</overview>
+              <doctitle>Amazon Ion Java ${project.version} API Reference</doctitle>
+              <windowtitle>Ion Java ${project.version}</windowtitle>
+              <header>Amazon Ion Java ${project.version} API Reference</header>
+              <encoding>UTF-8</encoding>
+              <bottom><![CDATA[<center>Copyright &#169; 2007&ndash;${build.year} Amazon.com. All Rights Reserved.</center>]]></bottom>
+              <name>Ion Java Javadoc</name>
+              <additionalparam>-Xdoclint:none</additionalparam>
+            </configuration>
+            <executions>
+              <execution>
+                <id>attach-javadocs</id>
+                <goals>
+                  <goal>jar</goal>
+                </goals>
+              </execution>
+            </executions>
+          </plugin>
+          <plugin>
+            <!-- GPG signing. -->
+            <groupId>org.apache.maven.plugins</groupId>
+            <artifactId>maven-gpg-plugin</artifactId>
+            <version>1.5</version>
+            <executions>
+              <execution>
+                <id>sign-artifacts</id>
```

Comment:
I see. It seems like we'd either have to all have Sonatype JIRA accounts, or share credentials (which doesn't seem acceptable).

Anyway, figuring that out doesn't hold up this PR.